src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/CollectionDataContract.cs

```diff
@@ -68,14 +68,16 @@ public T Value
 
     [DataContract(Namespace = "http://schemas.microsoft.com/2003/10/Serialization/Arrays")]
 #if USE_REFEMIT
-    public struct KeyValue<K, V>
+    public class KeyValue<K, V>
 #else
```

Comment:
The perf issue with dictionary is that they are handled as collection of KeyValue internally. As of now, KeyValue don't have default constructor so they're not created by the faster code path. I added a default constructor but since struct is not allowed to have explicit parameterless constructor, I've changed this type to class.